File: rollout_app/app/src/main/java/com/ammolite/rollout/ServerHandle.java
package com.ammolite.rollout;

import java.net.InetAddress;

public class ServerHandle {
    private InetAddress target;
    private String      name;

    public ServerHandle() { }

    public ServerHandle(InetAddress target, String name) {
        this.target = target;
        this.name = name;
    }

    public InetAddress getTarget() {
        return target;
    }

    public String getName() {
        return name;
    }

    public void setTarget(InetAddress target) {
        this.target = target;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return name + " - " + target.getHostAddress();
    }

    @Override
    public boolean equals(Object handle) {
        boolean same = false;

        if (handle != null && handle instanceof ServerHandle) {
            same = (name.equals(((ServerHandle)handle).name) && target.equals(((ServerHandle)handle).target));
        }

        return same;
    }
}

File: spherocontrol/ServerHandle.js
function ServerHandle(name, ip) {
    this._name = name;
    this._ip   = ip;
}

ServerHandle.prototype.getName = function() {
    return this._name;
};

ServerHandle.prototype.getIP = function() {
    return this._ip;
};

ServerHandle.prototype.toString = function() {
    return "\"" + this._name + "\" (" + this._ip + ":7777)";
}

module.exports = ServerHandle;